pkg/proxy/apis/config/types.go

```diff
@@ -302,3 +307,23 @@ func (m *LocalMode) Type() string {
 // NodePortAddressesPrimary is a special value for NodePortAddresses indicating that it
 // should only use the primary node IPs.
 const NodePortAddressesPrimary string = "primary"
+
+// IPFamilyPolicy controls the nodeIP detection logic, which further influences if LoadBalancer
+// services should be allowed on node when SourceRanges are set and on which interfaces node
+// ports should be exposed when nodePortAddresses is set to 'primary'.
+type IPFamilyPolicy string
+
+const (
+	// IPFamilyPolicySingleStack indicates that kube-proxy is required to work with NodeIP
+	// of single IP family. The NodeIP assigned is based on NodeIPOverride.
+	IPFamilyPolicySingleStack IPFamilyPolicy = "SingleStack"
+
+	// IPFamilyPolicyPreferDualStack indicates that kube-proxy is preferred to work with NodeIPs
+	// of both the IP families. The NodeIPs will be auto-detected if NodeIPOverride is empty
+	// or missing value for a particular IPFamily.
+	IPFamilyPolicyPreferDualStack IPFamilyPolicy = "PreferDualStack"
+
+	// IPFamilyPolicyRequireDualStack indicates that kube-proxy is preferred to work with NodeIPs
+	// of both the IP families.
+	IPFamilyPolicyRequireDualStack IPFamilyPolicy = "RequireDualStack"
+)
```

Comment:
You don't need to duplicate these here; you can just reuse `v1.IPFamilyPolicy` in the KubeProxyConfiguration.